src/renderer/utils.ts

```diff
@@ -0,0 +1,136 @@
+import {remote} from "electron";
+import {URL} from "url";
+import path from 'path';
+
+import {ST} from "./const";
+import en from "./en";
+import Config from "./Config";
+
+
+export function getPath() {
+  return path.join(remote.app.getPath('appData'), 'flipflip');
+}
+
+export function getFileName(url: string) {
+  return url.substring(url.lastIndexOf("/"));
+}
+
+export function getFileGroup(url: string) {
+  switch (getSourceType(url)) {
+    case ST.tumblr:
+      let tumblrID = url.replace(/https?:\/\//, "");
+      tumblrID = tumblrID.replace(/\.tumblr\.com\/?/, "");
+      return tumblrID;
+    case ST.reddit:
+      let redditID = url;
+      if (url.endsWith("/")) redditID = redditID.slice(0, url.lastIndexOf("/"));
+      redditID = redditID.substring(redditID.lastIndexOf("/") + 1);
+      return redditID;
+    case ST.local:
+      return url.substring(url.lastIndexOf("\\"));
```

Comment:
Does this only work on Windows? Whenever dealing with paths, you really need to get the path separator from Node.

Reply:
Derp.... done